lapack/testlapack/dorgr2.go

```diff
@@ -0,0 +1,56 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"testing"
+
+	"golang.org/x/exp/rand"
+)
+
+type Dorgr2er interface {
+	Dorgr2(m, n, k int, a []float64, lda int, tau []float64, work []float64)
+
+	Dgerqfer
+}
+
+func Dorgr2Test(t *testing.T, impl Dorgr2er) {
+	rnd := rand.New(rand.NewSource(1))
+	for _, m := range []int{2, 5, 20} {
```

Comment:
I would start from `k`:
```
for _, k := range []int{0, 1, 2, 3, 4, 5, 10} {
    for _, m := range []int{k, k+1, k+2, k+4, k+5} {
        for _, n := range []int{m, m+1, m+2, m+4, m+7}
```
If the test runtime is too long then remove for example the two `+4` sizes.

Reply:
Dgerqf panics if tau is not of length min(m,n). Your change above also panics for when k is not min(m,n) inside `a[l*lda+m-n+i] = 1` in Dorgr2